### PR TITLE
[nix-ci] Maintenance

### PR DESCRIPTION
**Pull request by vbgl**
A few improvements:

  - fix the build system of Unicoq
  - adds QuickChick
  - adds a convenience option to exclude Coq and Coq libraries from the dependencies.

### CODE REVIEW

**Review by maximedenes — APPROVED** [review]
Wonderful!

File: dev/ci/META [hunk 1]
```diff
@@ -0,0 +1,2 @@
+archive(native) = "unicoq.cmxa"
+plugin(native) = "unicoq.cmxs"
```
Comment (SkySkimmer) [inline dev/ci/META]: What's this?

File: dev/ci/unicoq-num.patch [hunk 2]
```diff
@@ -0,0 +1,44 @@
+commit f29bc64ee3d8b36758d17e1f5d50812e0c93063b
```
Comment (SkySkimmer) [inline dev/ci/unicoq-num.patch]: What stops this patch from going upstream?

Comment (vbgl) [reply dev/ci/unicoq-num.patch]: Nobody seems competent to write a proper install script for Windows:
https://github.com/unicoq/unicoq/pull/17#issuecomment-442302813

Comment (SkySkimmer) [reply dev/ci/unicoq-num.patch]: That PR is merged and seems to do the same thing as the patch, so why do we need the patch?

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
If dev/ci/META and dev/ci/unicoq-num.patch are nix specific then they should be moved to dev/ci/nix, otherwise please explain how they should be used.
Also the META should probably have a more specific name eg META.unicoq

File: dev/ci/unicoq-num.patch (hunk 2 above)
Comment (maximedenes) [reply dev/ci/unicoq-num.patch]: IIRC it's been reverted.

**Comment by vbgl** [comment]
I moved the files related to `Unicoq` to a dedicated directory.

**Review by SkySkimmer — APPROVED** [review]

**Review by ejgallego — APPROVED** [review]
Good from my part, I think that all the issues with Unicoq should be solved once we switch Nix to a Dune build. I am targeting the new Dune verison for next week.

File: dev/ci/nix/README.md [hunk 3]
```diff
@@ -17,3 +17,10 @@ build-system of that project: `configure`, `make`, and `clean`. Therefore, after
 changing the working directory to the root of the sources of that project, the
 contents of these variables can be evaluated to respectively set-up, build, and
 clean the project.
+
+## Variant: nocoq
+
+The dependencies of the third-party developments are split into `buildInputs`
+and `coqBuildInputs`. The second list gathers the Coq libraries. In case you
+only want the non-coq dependencies (because you want to use Coq from you `PATH`),
```
Comment (Zimmi48) [inline dev/ci/nix/README.md]:
```suggestion
only want the non-coq dependencies (because you want to use Coq from your `PATH`),
```

**Review by Zimmi48 — COMMENTED** [review]
`NOCOQ` sounds useful

**Pull request merged**